lib/puppet/functions/new.rb

```diff
@@ -26,6 +26,6 @@ def new_function_for_type(t, scope)
   end
 
   def assert_type(type, value)
-    Puppet::Pops::Types::TypeAsserter.assert_instance_of('new():', type, value)
+    Puppet::Pops::Types::TypeAsserter.assert_instance_of("Converted value from #{type.to_s}.new()", type, value)
```

Comment:
Better to use:
```
Puppet::Pops::Types::TypeAsserter.assert_instance_of(['Converted value from %s.new()', type], type, value)
```
to avoid unnecessary formatting when value is an instance of type.